blueman/gui/GenericList.py

```diff
@@ -16,31 +16,26 @@ def _load(self, data):
         self.ids = {}
         self.columns = {}
 
-        types = []
-        for i in range(len(data)):
-            types.append(data[i][1])
-
-        types = tuple(types)
+        types = [row["type"] for row in data]
 
         self.liststore = Gtk.ListStore(*types)
         self.set_model(self.liststore)
 
-        for i in range(len(data)):
-
-            self.ids[data[i][0]] = i
-
-            if len(data[i]) == 4 or len(data[i]) == 5:
+        for i, row in enumerate(data):
+            self.ids[row["id"]] = i
 
-                column = Gtk.TreeViewColumn(data[i][3])
+            if "renderer" not in row:
+                continue
 
-                column.pack_start(data[i][2], True)
-                column.set_attributes(data[i][2], **data[i][3])
+            column = Gtk.TreeViewColumn(row["renderer"])
```

Comment:
Lol, the first argument to the Gtk.TreeViewColumn is being used to set the title. I'll just drop the argument as we do not use the title anywhere.